File: data/make-pages.js

const letters = [ 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'M', 'N', 'O', 'P', 'R', 'S', 'T', 'W', 'Y', 'Z' ];

const fortnight = 1000 * 60 * 60 * 24 * 14; // milliseconds in 2 weeks

let date = new Date("2021-10-26T04:00:00.000+02:00");
let id = 154;

letters.forEach((l) => {
    console.log(`{ "generation": 8, "id":${id++}, "letter": "${l}", "open": true, "startDate": "${date.toISOString()}" },`);
    date = new Date(date.getTime() + fortnight);
});

File: data/makepages.pl
#!/usr/bin/env perl

use strict;
use warnings;
use POSIX qw(mktime strftime);

use constant TWO_WEEKS => 60 * 60 * 24 * 14; # Get ready for a surprise!

my ($sec, $min, $hour, $mday, $mon, $year) = (0, 0, 0, 25, 5, 124); # 2024-06-25T00:00:00.000+02:00
my $date = mktime($sec, $min, $hour, $mday, $mon, $year);

my $id = 224;
my $generation = "iii";

for ("A" .. "Z") {
    next if /[IQUVZ]/;

    $date += TWO_WEEKS;
	my $datestr = strftime("%Y-%m-%dT00:00:00.000%z", localtime $date);
    print <<"    _end_";
      },
      {
        "generation": "$generation",
        "id": $id,
        "letter": "$_",
        "open": true,
        "startDate": "$datestr"
    _end_

    $id++;
}



